Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp

```diff
@@ -161,7 +161,7 @@ void WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction(const Navigat
     // Notify the UIProcess.
     if (policyDecisionMode == PolicyDecisionMode::Synchronous) {
 #if PLATFORM(COCOA)
-        if (!linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::AsyncFragmentNavigationPolicyDecision)) {
+        if (navigationAction.userGestureToken() || !linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::AsyncFragmentNavigationPolicyDecision)) {
```

Comment:
I wonder if this should check for `processingUserGesture()` instead of  `userGestureToken()`...

The former is a bit more restrictive, and (seems to) take gestures that only trigger media playback into account.